src/storage.ts

```diff
@@ -362,11 +367,13 @@ export function createStorage<T extends StorageValue>(
           ...maskedMounts.filter((p) => !p.startsWith(mount.mountpoint)),
         ];
       }
-      return base
-        ? allKeys.filter(
-            (key) => key.startsWith(base!) && key[key.length - 1] !== "$"
-          )
-        : allKeys.filter((key) => key[key.length - 1] !== "$");
+      return allKeys
+        .filter((key) => filterKeyByDepth(key, opts.depth))
+        .filter(
+          base
```

Comment:
perf: I wish somehow we could merge filters. Current change adds one filter iteration even for users not setting `depth`